exiftool.go

```diff
@@ -84,16 +85,47 @@ func NewExiftool(opts ...func(*Exiftool) error) (*Exiftool, error) {
 	if err = e.cmd.Start(); err != nil {
 		return nil, fmt.Errorf("error when executing command: %w", err)
 	}
+	if err := e.log("executing shell command:\n\t", e.cmd.String()); err != nil {
+		return nil, err
+	}
 
 	return &e, nil
 }
 
+func (e *Exiftool) log(args ...interface{}) error {
+	for _, w := range e.debugWriters {
+
+		_, err := fmt.Fprintln(w, args...)
+
+		if err != nil {
+			return err
```

Comment:
What about telling that something went wrong during the "logging" process ?
An error raised when calling `fmt.Fprintln` to log won't be distinguishable from an error raised by the `fmt.Fprintln` that controls exiftool (providing arguments).
What do you think about adding an `return fmt.Errorf("error while logging arguments: %w", err)` ?

Reply:
Ya, it would be better to add some context. I'd suggest a slightly different message, something like `fmt.Errorf("unable to write logs (writerType=%T;args=%v): %w", w, args, err)`. How does that sound to you?

Reply:
That's perfect !

Reply:
@jnathanh , do you want me to make the changes ?